File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution(object):
    
    
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        jumps = 0
        target = len(nums)-1
        farthest = 0
        current_end = 0
        for i in range(0, len(nums)-1):
            farthest = max(i+nums[i], farthest)

            if i==current_end:
                jumps+=1
                current_end = farthest
            print(f"{i=} {current_end=} {farthest=} {jumps=}")

        return jumps

            

File: 0045-jump-game-ii/0045-jump-game-ii.scala
object Solution {
      def jump(nums: Array[Int]): Int = {


    var current_end = 0
    var farthest = 0
    var jump = 0
    for ( i<-0 until nums.length-1){
      
      farthest = scala.math.max(farthest, nums(i)+i)
      
      if(i==current_end){
        jump+=1
        current_end = farthest
      }
    
      
    }
    
    jump
  }
}